### PR TITLE
fix: Ensure unwindConditionalToModel generates valid HCL

### PR DESCRIPTION
**Pull request by biblicalph**
During QA of filter processor, a bug was discovered in the unwindConditionalToModel function; it sometimes generates invalid HCL - expressions and expressions_group are both set at a given level under some circumstances.
The issue is addressed here using test data including the QA data.

Ref: LOG-18633

### CODE REVIEW

File: internal/provider/models/modelutils/conditional_types.go [hunk 1]
```diff
@@ -125,3 +128,126 @@ func GetAttributesByLevel(level int) map[string]schema.Attribute {
 func GetChildExpressionGroupTypeByLevel(level int) attr.Type {
 	return childExpressionGroups[level].Type()
 }
+
+func UnwindConditionalToModel(component map[string]any) ObjectValue {
```
Comment (biblicalph) [inline internal/provider/models/modelutils/conditional_types.go:132]: Moved out of reduce processor and made a couple of modifications to ensure valid HCL is generated

File: internal/provider/models/modelutils/conditional_types.go [hunk 2]
```diff
@@ -125,3 +128,89 @@ func GetAttributesByLevel(level int) map[string]schema.Attribute {
...
+		} else if attributeTypes["expressions_group"] != nil {
+			attributeValues["expressions_group"] = NewListNull(expressionGroupType)
+		}
+
+		rootHasExpressionsAndGroups := len(leafItems) > 0 && len(groupItems) > 0
+		if !rootHasExpressionsAndGroups {
+			return NewObjectValueMust(attributeTypes, attributeValues), true
+		}
+
+		// prepend leaf item expressions to the expressions group
```
Comment (jorgebay) [inline internal/provider/models/modelutils/conditional_types.go:180]: IIUC the logic should be: "When there's a mix of expressions and expression groups, convert the expressions to expression groups"

Reading the code, that logic is not very clear and there's repeated code.

Can we detect early if there are both `expressions` and `expression_groups` as child, convert them in place and add them to the groups items?

**Review by jorgebay — COMMENTED** [review]
Nice coverage!

I've added some remarks below. I think we can detect the expected structure earlier and simplify the logic a little bit.

**Review by jorgebay — APPROVED** [review]
Nice!

File: internal/provider/models/modelutils/conditional_types.go (hunk 1 above)
Comment (darinspivey) [reply internal/provider/models/modelutils/conditional_types.go:130]: Thanks!  LGTM

**Pull request merged**